app/controllers/searches_controller.rb

```diff
@@ -11,10 +11,56 @@ def index
       end
 
       render :index
-    # else
-    #   flash.now[:alert] = 'Please enter a city name and select a type to search'
-    #   @places = []
-    #   # render :index
+      # else
+      #   flash.now[:alert] = 'Please enter a city name and select a type to search'
+      #   @places = []
+      #   # render :index
     end
   end
-end
\ No newline at end of file
+
+  def add_to_favorites
+    place_id = params[:place][:id].to_i
+    place = params[:place][:place_type].singularize
+
+    case place
+    when "Attraction"
+      if current_user.attraction_ids.include?(place_id)
+        flash[:alert] = failure_message(place)
+      else
+        current_user.attractions << Attraction.find(place_id)
+        flash[:notice] = success_message(place)
+      end
```

Comment:
This is more expensive than it needs to be. `current_user.attraction_ids.include?` queries the database for the existing IDs, and then `Attraction.find(place_id)` makes another query to the database for that record. You should be able to do `current_user.attractions.create(attraction_id: place_id)` and then `rescue ActiveRecord::RecordNotUnique`.

There's a separate question of whether you even want to display an error message when somebody tries to add a place that they already have, because in both cases, when they go to the dashboard, the place is there.

Reply:
When I try `current_user.attractions.create(attraction_id: place_id)`

I get the error: `unknown attribute 'attraction_id' for Attraction.`

If I try `current_user.attractions.create(Attraction.find(1))`

```
ArgumentError: When assigning attributes, you must pass a hash as an argument, Attraction passed.
	from /home/racoder/codeTheDream/firefish-practicum/app/controllers/places_controller.rb:8:in `add'

```

I can't get stuck on this now so I'll have to go through with what worked